spec/models/ems_event_spec.rb

```diff
@@ -472,42 +472,42 @@
   end
 
   context 'manager refresh' do
-    let(:ems_cloud) { FactoryBot.create(:ems_cloud) }
-    let(:ems_event_cloud) {
+    let(:ems)       { FactoryBot.create(:ems_cloud) }
+    let(:ems_event) do
       FactoryBot.create(
         :ems_event,
-        :ext_management_system => ems_cloud,
-        :event_type => "CloneVM_Task",
-        :full_data  => {"info" => {"task" => "task-5324"}}
+        :ext_management_system => ems,
+        :event_type            => "CloneVM_Task",
+        :full_data             => {"info" => {"task" => "task-5324"}}
       )
-    }
+    end
 
-    let(:ems_infra) { FactoryBot.create(:ems_infra) }
-    let(:ems_event_infra) {
-      FactoryBot.create(
-        :ems_event,
-        :ext_management_system => ems_infra,
-        :event_type => "CloneVM_Task",
-        :full_data  => {"info" => {"task" => "task-5324"}}
-      )
-    }
+    context "targeted refresh supported" do
+      let(:target_parser) { double("EventTargetParser") }
 
-    it 'performs a targeted refresh if supported' do
-      klass = ems_cloud.class.const_get('EventTargetParser')
-      target_parser = klass.new(ems_cloud)
+      before do
+        allow(ems).to receive(:allow_targeted_refresh?).and_return(true)
+        allow(ems.class.const_get('EventTargetParser')).to receive(:new).and_return(target_parser)
```

Comment:
NOTE I don't love this but it is how the existing test stubbed the EventTargetParser